lib/_http_client.js

```diff
@@ -562,7 +562,13 @@ function tickOnSocket(req, socket) {
   socket.on('close', socketCloseListener);
 
   if (req.timeout) {
-    socket.once('timeout', () => req.emit('timeout'));
+    const emitRequestTimeout = () => req.emit('timeout');
+    socket.once('timeout', emitRequestTimeout);
+    req.on('response', (r) => {
```

Comment:
hm, I wonder if the response event is emitted in all cases if the request is aborted...will have to dig into that